pkg/metrics/metrics.go

```diff
@@ -0,0 +1,129 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
```

Comment:
This is confusing. I will keep it clean and only define the variable types. Also I dont think this is the place to do this because this class should only define what can you do with the `metrics` package, not how to call it.
It should be:
```
var (
	upStatus                                                                    prometheus.Gauge
	buildCount, buildRunCount, buildRunEstablishDuration, buildRunReadyDuration prometheus.GaugeVec
	buildRunReadyDurationHis                                                    prometheus.HistogramVec
)
```

and should be defined somewhere else not here. Wherever you wanna call the pkg.

Reply:
I suggest we keep this style first until we finalize them in future phase 2.

For example, `buildCount`, `buildRunCount` may change to `COUNT` type just increase the count.